### PR TITLE
SRCH-3671 Add Jest and react-testing-library

### PR DESCRIPTION
**Pull request by ddzz**
## Summary
This PR adds Jest and react-testing-library. Three tests have also been added for the SERP redesign React component.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: app/javascript/components/ResultsPage.tsx [hunk 1]
```diff
@@ -111,9 +111,9 @@ class ResultsPage extends React.Component<ResultsPageProps, Nothing> {
         </GridContainer>
 
         <div id='results'>
-          {this.props.results.map(result => {
+          {this.props.results.map((result, index) => {
             return (
-              <div className='result'>
+              <div className='result' key={index}>
```
Comment (ddzz) [inline app/javascript/components/ResultsPage.tsx:116]: This key was added to fix a React warning for unique keys when iterating over an array.

**Review by lorawoodford — CHANGES_REQUESTED** [review]
This all looks good, but can I request you add a simple one liner to the [testing section of the readme](https://github.com/GSA/search-gov#tests) noting the existence of these tests/how to run them?  Might as well do it now with these first three passing tests so we don't forget.

**Review by lorawoodford — DISMISSED** [review]
LGTM

**Review by lorawoodford — APPROVED** [review]
LGTM

**Pull request merged**